src/extension/configuration.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as vscode from "vscode";
+
+/**
+ * Gets the path to the Bazel executable specified by the workspace
+ * configuration, if present.
+ *
+ * @returns The path to the Bazel executable specified in the workspace
+ * configuration, or just "bazel" if not present (in which case the system path
+ * will be searched).
+ */
+export function getDefaultBazelExecutablePath(): string {
+  // Try to retrieve the executable from VS Code's settings. If it's not set,
+  // just use "bazel" as the default and get it from the system PATH.
+  const bazelConfig = vscode.workspace.getConfiguration("bazel");
+  const bazelExecutable = bazelConfig.executable as string;
+  if (bazelExecutable.length === 0) {
+    return "bazel";
```

Comment:
I guess this also fixes #88?  Not sure if the suggested api there is any better.

Reply:
I didn't make any changes to this function, I just moved it around. So if it does fix it then it was fixed before. Since this bug also references buildifier I think we should leave it open until that is also addressed.

Reply:
Yeah, I'm not sure how #88 is happening since we explicitly have code that avoids the empty string case. It's worth looking into, but separately.